src/main/java/org/opensearch/timeseries/NodeStateManager.java

```diff
@@ -268,8 +277,10 @@ private ActionListener<GetResponse> onGetConfigResponse(
                     return;
                 }
 
-                NodeState state = states.computeIfAbsent(configID, configId -> new NodeState(configId, clock));
-                state.setConfigDef(config);
+                if (cache) {
```

Comment:
at this point didn't we already make the request to get the config, so fetching from cache should happen earlier?

Reply:
cache here means whether we should put the get config response to cache. Fetching from cache happens in the caller of this method.